### ISSUE TITLE
About the official way of testing the model trained by openfold

### ISSUE THREAD

**Issue by shiyemin** [aqlaboratory/openfold#88]
Our team has tried and trained several settings to train openfold. Our inference code is modified based on run_pretrained_openfold.py by replacing import_jax_weights_ with torch.load.
However, no model can produce proper prediction and everything looks like random.

Is there a official code to inference based on openfold's model?

**Comment by gahdritz** [comment]
Do the metrics look reasonable during training? Do the predictions you get out just superimpose all of the amino acids onto a single point?

**Comment by akiladonz** [comment]
@shiyemin could you please share your version of the inference code for the reference purposes.

**Comment by shiyemin** [comment]
@gahdritz The training metric seems reasonable. Both with and without DeepSpeed are tested, and none of them work. The points did not superimpose.
@akiladonz Here is our inference code for you reference.
# for ddp
```
def main(args):
    config = model_config(args.model_name)
    model = AlphaFold(config)
    model = model.eval()

    # import_jax_weights_(model, args.param_path, version=args.model_name)
    
    model_file = 'PATH/TO/checkpoints/epoch=3-step=9999.ckpt'
    sd = torch.load(model_file, map_location='cpu')
    sd = sd['state_dict']
    state_dict = {k[len("model."):]:v for k,v in sd.items()}
    model.load_state_dict(state_dict)

    model = model.to(args.model_device)
```

# for deepspeed
```
def main(args):
    config = model_config(args.model_name)
    model = AlphaFold(config)
    model = model.eval()

    # import_jax_weights_(model, args.param_path, version=args.model_name)

    model_file = 'PATH/TO/checkpoints/epoch=7-step=2503.ckpt/global_step2504/mp_rank_00_model_states.pt'
    sd = torch.load(model_file, map_location='cpu')
    sd = sd['state_dict']
    state_dict = {k[len("model."):]:v for k,v in sd.items()}
    model.load_state_dict(state_dict)
    
    model = model.to(args.model_device)
```

**Comment by gahdritz** [comment]
For the DeepSpeed one first, try the following:

```python3
from pytorch_lightning.utilities.deepspeed import convert_zero_checkpoint_to_fp32_state_dict

# Load model, train, etc.
# ...

checkpoint_path = 'PATH/TO/checkpoints/epoch=7-step=2503.ckpt
out_path = "converted_checkpoint.pt"

convert_zero_checkpoint_to_fp32_state_dict(checkpoint_path, out_path)
loaded_checkpoint = torch.load(out_path)
model.load_state_dict(d["ema"]["params"])
```
I also had trouble loading weights before I found this approach, but this works for me.

**Comment by shiyemin** [comment]
@gahdritz Can you share your openfold model (even trained on small-scale dataset)?

**Comment by lhatsk** [comment]
Just fyi, I don't have any problems loading the parameters directly:

sd = torch.load('mp_rank_00_model_states.pt')
model.load_state_dict(sd['ema']['params'])

@shiyemin curious what performance you get in terms of lddt-CA on validation and for how long/ on how many samples you train.

**Comment by gahdritz** [comment]
@shiyemin was your issue resolved?
